src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1653,6 +1653,41 @@ void C2_MacroAssembler::round_double_mode(FloatRegister dst, FloatRegister src,
   bind(done);
 }
 
+// According to Java SE specification, for floating-point signum operations, if
+// on input we have NaN or +/-0.0 value we should return it,
+// otherwise return +/- 1.0 using sign of input.
+// tmp1 - used to store result of fclass operation,
+// one - gives us a floating-point 1.0 (got from matching rule)
+// bool single_precision - specififes single or double precision operations will be used.
+void C2_MacroAssembler::signum_fp(FloatRegister dst, FloatRegister src, FloatRegister one, Register tmp1, bool is_double) {
+  assert_different_registers(dst, src, one);
```

Comment:
This constraint could be relaxed if we use scratch register `t0` here instead of `tmp`.